Src/Support/Google.Apis.Auth/OAuth2/AwsExternalAccountCredential.AwsSignedSubjectToken.cs

```diff
@@ -0,0 +1,279 @@
+﻿/*
+Copyright 2022 Google LLC
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    https://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+using Google.Apis.Util;
+using Newtonsoft.Json;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Net.Http;
+using System.Runtime.CompilerServices;
+using System.Security.Cryptography;
+using System.Text;
+
+namespace Google.Apis.Auth.OAuth2
+{
+    public partial class AwsExternalAccountCredential
+    {
+        /// <summary>
+        /// Partial implementation of AWS signed request, enough to support signing
+        /// of a, usually, AWS GetCallerIdentity request. The signed request is sent
+        /// to Google's subject token service as the subject token to be exchanged for
+        /// access tokens. Google STS triggers the request as specified by the signed
+        /// request to verify the callers identity.
+        /// </summary>
+        private sealed class AwsSignedSubjetToken
+        {
+            private const string StVerificationHttpMethod = "POST";
+            private const string BasicFormatIso8601 = "yyyyMMddTHHmmssZ";
+            private const string ShortDateFormat = "yyyyMMdd";
+            private const string AwsRequestType = "aws4_request";
+            private const string AwsSha256Designation = "AWS4-HMAC-SHA256";
+
+            [JsonProperty("url")]
+            public string Url { get; }
+
+            [JsonProperty("method")]
+            public string HttpMethod { get; }
+
+            [JsonProperty("body")]
+            public string Body { get; }
+
+            [JsonProperty("headers")]
+            public AwsSignedSubjectTokenHeader[] Headers { get; }
+
+            private AwsSignedSubjetToken(string url, string httpMethod, string body, AwsSignedSubjectTokenHeader[] headers)
```

Comment:
Possibly document that we construct this manually, but then serialize it? (Otherwise it's slightly odd to have JsonProperty attributes but no public parameterless constructor.)